src/Microsoft.ML.CodeGen/CodeGenerator/CSharp/CodeGenerator.cs

```diff
@@ -218,6 +218,10 @@ internal IList<string> GenerateClassLabels()
             IList<string> result = new List<string>();
             foreach (var column in _columnInferenceResult.TextLoaderOptions.Columns)
             {
+                if (IsColumnIgnored(column.Name))
+                {
+                    continue;
```

Comment:
--summary of non-github discussion--

I'd recommend leaving the generation of the unused (Ignore) column in place. I think what you're purposing is "ignore the entire existence of the column" vs. the current "just don't use it for featurization".

Generally column are ignored by the AutoML heuristics. These, while generally good, are imperfect. This leads to AutoML ignore column which could be useful.

Removing the column from the input class makes the column no longer appear in the IDataView, which means users won't be able to easily add back the missing column. 

When the user explicitly ignores a column, it's often done due to the column leaking the label. When ignoring a column from the CLI, after CodeGen, the user can handle this column without leaking.

Reply:
From a user's perspective, I find it surprising that these columns still make it into the code generation when I told the tool to exclude this column. My opinion is we should be removing those columns.

Maybe it would be best to have an option to include/exclude these columns or not. That way the tool or user can decide whether they want these columns generated or not.